### PR TITLE
Add tests and nil checks

### PR DESCRIPTION
**Pull request by TunedMystic**

### CODE REVIEW

File: dom/dom_test.go [hunk 1]
```diff
@@ -70,6 +70,70 @@ func Test_Group_Render_Errors(t *testing.T) {
...
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var w io.Writer = io.Discard
+			if tt.writer != nil {
+				w = tt.writer
+			}
+			err := tt.node.Render(w)
+			assert.Error(t, err, tt.expectedErr)
```
[the remark this answers is not in the document]
Comment (TunedMystic) [reply dom/dom_test.go:132]: I am using an internal assert package, so this does not apply.

File: dom/dom_test.go [hunk 2]
```diff
@@ -96,6 +160,32 @@ func Test_HtmlString(t *testing.T) {
...
+	t.Run("Render error", func(t *testing.T) {
+		r := Doctype(Html(Lang("en"), Head(Meta(Charset("UTF-8")))))
+		errWriterSentinel := errors.New("writer error from htmlDoctype test")
+		writer := &errorWriter{
+			targetErr:      errWriterSentinel,
+			failOnNthWrite: 1,
+		}
+
+		err := r.Render(writer)
+		assert.Error(t, err, errors.New("writer error from htmlDoctype test"))
```
[the remark this answers is not in the document]
Comment (TunedMystic) [reply dom/dom_test.go:179]: I am using an internal assert package, so this does not apply.

File: dom/dom_test.go [hunk 3]
```diff
@@ -135,6 +225,73 @@ func Test_ControlStructures(t *testing.T) {
...
+// ------------------------------------------------------------------
+
+func Test_Rendering_NilChecks(t *testing.T) {
+	t.Run("htmlElement attribute is (*htmlAttr)(nil)", func(t *testing.T) {
+		var nilConcreteAttribute Node = (*htmlAttr)(nil)
+		element := Div(Class("test"), nilConcreteAttribute, Id("my-id"))
+
+		var sb strings.Builder
+		err := element.Render(&sb)
+		assert.Error(t, err, nil)
```
[the remark this answers is not in the document]
Comment (TunedMystic) [reply dom/dom_test.go:241]: I am using an internal assert package, so this does not apply.

File: dom/dom_test.go [hunk 4]
```diff
@@ -135,6 +225,73 @@ func Test_ControlStructures(t *testing.T) {
...
+		assert.Equal(t, sb.String(), `<div class="test" id="my-id"></div>`)
+	})
+
+	t.Run("htmlElement child node is (Node)(nil)", func(t *testing.T) {
+		var nilInterfaceNode Node = nil
+		element := Div(Text("hello"), nilInterfaceNode, Text("world"))
+
+		var sb strings.Builder
+		err := element.Render(&sb)
+		assert.Error(t, err, nil)
```
[the remark this answers is not in the document]
Comment (TunedMystic) [reply dom/dom_test.go:251]: I am using an internal assert package, so this does not apply.

File: dom/dom_test.go [hunk 5]
```diff
@@ -135,6 +225,73 @@ func Test_ControlStructures(t *testing.T) {
...
+		var nilInterfaceNode Node = nil
+		group := Group{
+			Text("first"),
+			nilInterfaceNode,
+			Text("third"),
+		}
+
+		var sb strings.Builder
+		err := group.Render(&sb)
+		assert.Error(t, err, nil)
```
[the remark this answers is not in the document]
Comment (TunedMystic) [reply dom/dom_test.go:265]: I am using an internal assert package, so this does not apply.

File: dom/dom_test.go [hunk 6]
```diff
@@ -135,6 +225,73 @@ func Test_ControlStructures(t *testing.T) {
...
+		mapper := Map(items, func(item string) Node {
+			if item == "two" {
+				return nil
+			}
+			return Text(item)
+		})
+
+		var sb strings.Builder
+		err := mapper.Render(&sb)
+		assert.Error(t, err, nil)
```
[the remark this answers is not in the document]
Comment (TunedMystic) [reply dom/dom_test.go:280]: I am using an internal assert package, so this does not apply.

File: dom/dom_test.go [hunk 7]
```diff
@@ -135,6 +225,73 @@ func Test_ControlStructures(t *testing.T) {
...
+		assert.Equal(t, sb.String(), `onethree`)
+	})
+
+	t.Run("htmlDoctype sibling is (Node)(nil)", func(t *testing.T) {
+		var nilSiblingNode Node = nil
+		doctypeNode := Doctype(nilSiblingNode)
+
+		var sb strings.Builder
+		err := doctypeNode.Render(&sb)
+		assert.Error(t, err, nil)
```
[the remark this answers is not in the document]
Comment (TunedMystic) [reply dom/dom_test.go:290]: I am using an internal assert package, so this does not apply.
